core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -174,6 +206,11 @@ lib_pthread_init()
         os_mutex_destroy(&thread_global_lock);
         return false;
     }
+    if (!(sem_info_map = bh_hash_map_create(32, true, (HashFunc)sem_name_hash,
+                                            thread_handle_equal, NULL,
```

Comment:
Should not use thread_handle_equal, here it is to check whether two keys are equal, or whether two strings are equal, refer to `const_str_set_insert` in aot_loader.c.